Compiled from "Promise.scala"
public final class scala.concurrent.impl.Promise$ {
  public static final scala.concurrent.impl.Promise$ MODULE$;

  public static {};
    Code:
       0: new           #2                  // class scala/concurrent/impl/Promise$
       3: invokespecial #12                 // Method "<init>":()V
       6: return

  public <T> scala.util.Try<T> scala$concurrent$impl$Promise$$resolveTry(scala.util.Try<T>);
    Code:
       0: aload_1
       1: instanceof    #16                 // class scala/util/Failure
       4: ifeq          24
       7: aload_1
       8: checkcast     #16                 // class scala/util/Failure
      11: astore_2
      12: aload_0
      13: aload_2
      14: invokevirtual #20                 // Method scala/util/Failure.exception:()Ljava/lang/Throwable;
      17: invokespecial #24                 // Method resolver:(Ljava/lang/Throwable;)Lscala/util/Try;
      20: astore_3
      21: goto          26
      24: aload_1
      25: astore_3
      26: aload_3
      27: areturn

  private <T> scala.util.Try<T> resolver(java.lang.Throwable);
    Code:
       0: aload_1
       1: instanceof    #31                 // class scala/runtime/NonLocalReturnControl
       4: ifeq          28
       7: aload_1
       8: checkcast     #31                 // class scala/runtime/NonLocalReturnControl
      11: astore_2
      12: new           #33                 // class scala/util/Success
      15: dup
      16: aload_2
      17: invokevirtual #37                 // Method scala/runtime/NonLocalReturnControl.value:()Ljava/lang/Object;
      20: invokespecial #40                 // Method scala/util/Success."<init>":(Ljava/lang/Object;)V
      23: astore        6
      25: goto          147
      28: aload_1
      29: instanceof    #42                 // class scala/util/control/ControlThrowable
      32: ifeq          65
      35: aload_1
      36: checkcast     #42                 // class scala/util/control/ControlThrowable
      39: astore_3
      40: new           #16                 // class scala/util/Failure
      43: dup
      44: new           #44                 // class java/util/concurrent/ExecutionException
      47: dup
      48: ldc           #46                 // String Boxed ControlThrowable
      50: aload_3
      51: checkcast     #48                 // class java/lang/Throwable
      54: invokespecial #51                 // Method java/util/concurrent/ExecutionException."<init>":(Ljava/lang/String;Ljava/lang/Throwable;)V
      57: invokespecial #54                 // Method scala/util/Failure."<init>":(Ljava/lang/Throwable;)V
      60: astore        6
      62: goto          147
      65: aload_1
      66: instanceof    #56                 // class java/lang/InterruptedException
      69: ifeq          101
      72: aload_1
      73: checkcast     #56                 // class java/lang/InterruptedException
      76: astore        4
      78: new           #16                 // class scala/util/Failure
      81: dup
      82: new           #44                 // class java/util/concurrent/ExecutionException
      85: dup
      86: ldc           #58                 // String Boxed InterruptedException
      88: aload         4
      90: invokespecial #51                 // Method java/util/concurrent/ExecutionException."<init>":(Ljava/lang/String;Ljava/lang/Throwable;)V
      93: invokespecial #54                 // Method scala/util/Failure."<init>":(Ljava/lang/Throwable;)V
      96: astore        6
      98: goto          147
     101: aload_1
     102: instanceof    #60                 // class java/lang/Error
     105: ifeq          137
     108: aload_1
     109: checkcast     #60                 // class java/lang/Error
     112: astore        5
     114: new           #16                 // class scala/util/Failure
     117: dup
     118: new           #44                 // class java/util/concurrent/ExecutionException
     121: dup
     122: ldc           #62                 // String Boxed Error
     124: aload         5
     126: invokespecial #51                 // Method java/util/concurrent/ExecutionException."<init>":(Ljava/lang/String;Ljava/lang/Throwable;)V
     129: invokespecial #54                 // Method scala/util/Failure."<init>":(Ljava/lang/Throwable;)V
     132: astore        6
     134: goto          147
     137: new           #16                 // class scala/util/Failure
     140: dup
     141: aload_1
     142: invokespecial #54                 // Method scala/util/Failure."<init>":(Ljava/lang/Throwable;)V
     145: astore        6
     147: aload         6
     149: areturn

  private scala.concurrent.impl.Promise$();
    Code:
       0: aload_0
       1: invokespecial #65                 // Method java/lang/Object."<init>":()V
       4: aload_0
       5: putstatic     #67                 // Field MODULE$:Lscala/concurrent/impl/Promise$;
       8: return
}
